src/Getters/Parsha.php

```diff
@@ -51,12 +51,17 @@ private function parshios($index, $format = null)
     /**
      * Get the current week's Parsha.
      *
-     * @return string
+     * @return string|null
      */
     private function parshasHashavua($galus = null)
     {
         $galus = $this->getGalusMode($galus);
 
+        // Yuntif does not have a parsha
+        if ($this->comingShabbos()->isYuntif() || $this->comingShabbos()->isCholHamoed()) {
```

Comment:
This just checks if this coming Shabbos is Yuntif or Chol Hamoed. However, if today is Yuntif or Chol Hamoed, I assume we also don't want to have a Parsha (except Simchas Torah)?

Reply:
They way the `parsha` works is always to get that week's parsha, rather than the parsha of the current day. On a random Wednesday the only applicable parsha is that of the coming shabbos, so I see no difference if it happened to be yuntif on that wednesday

Reply:
After thinking about this a little more, I think that you shouldn't check if this coming Shabbos is Yuntif or Chol Hamoed. Instead you should check if today is Shabbos, and if not then return the Parsha for this coming Shabbos. Then the next check should just be if today is Yuntif or Chol Hamoed. The bonus is that you can get rid of the helper method and won't potentially have to create two more variables that are immediately orphaned. Granted, this is nit-picky, but I think it makes it clearer when looking at the code what it is doing